standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnHandler.java

```diff
@@ -369,11 +369,15 @@ public void setConf(Configuration conf){
 
     int maxPoolSize = MetastoreConf.getIntVar(conf, ConfVars.CONNECTION_POOLING_MAX_CONNECTIONS);
     synchronized (TxnHandler.class) {
-      if (connPool == null) {
-        connPool = setupJdbcConnectionPool(conf, maxPoolSize);
-      }
-      if (connPoolMutex == null) {
-        connPoolMutex = setupJdbcConnectionPool(conf, maxPoolSize);
+      try (DataSourceProvider.DataSourceNameConfigurator configurator =
+               new DataSourceProvider.DataSourceNameConfigurator(conf, "txn")) {
```

Comment:
nit: should we call this txnhandler instead of txn?

Reply:
sure